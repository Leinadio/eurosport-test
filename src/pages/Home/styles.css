.container {
    display: flex;
    justify-content: center;
}

/*@media all and (min-width: 768px) {*/
/*    .container {*/
/*        width: 320px;*/
/*        margin: auto;*/
/*    }*/
/*}*/

.home-title {
    font-size: 24px;
    font-family: Arial;
    margin-bottom: 24px;
    text-align: center;
    margin-top: 32px;
}

.cards {
    display: flex;
    flex-direction: column;
}

@media all and (min-width: 768px) {
    .cards {
        display: flex;
        flex-direction: row;
    }
}
