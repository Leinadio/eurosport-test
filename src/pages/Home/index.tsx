import React from 'react';
import { useQuery } from '@apollo/client';
import { HEAD_TO_HAD, HeadToHeadType } from '../../services/graphql/headToHead';
import Loading from '../../components/Loading';
import Error from '../../components/Error';
import Card from '../../components/Card';
import { generateRandomNumber } from '../../utils/generateRandomNumber';
import './styles.css'

function displayCards(data: Array<HeadToHeadType>): Array<JSX.Element> {
  return data.map((el: HeadToHeadType): JSX.Element => (
    <Card
      key={`${el.firstname}${generateRandomNumber()}`}
      url={el.picture.url}
      firstname={el.firstname}
      lastname={el.lastname}
      shortname={el.shortname}
      sex={el.sex}
      age={el.stats.age}
      height={el.stats.height}
      weight={el.stats.weight}
      points={el.stats.points}
      rank={el.stats.rank}
      last={el.stats.last}
      country={el.country}
    />
  ))
}

function DisplayLoadingOrErrorOrData(): JSX.Element {
  const { loading, error, data } = useQuery(HEAD_TO_HAD);
  if (loading) {
    return (
      <Loading />
    )
  }
  if (error || !data) {
    return (
      <Error />
    )
  }
  return (
    <div className="container">
      <div className="cards">
        {displayCards(data.headToHead)}
      </div>
    </div>
  )
};

export default function Home(): JSX.Element {
  return (
    <div>
      <h1 className="home-title">Players</h1>
      {DisplayLoadingOrErrorOrData()}
    </div>
  )
}
