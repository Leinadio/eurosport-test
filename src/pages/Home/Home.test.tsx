import React from 'react';
import renderer from 'react-test-renderer';
import Home from './'

jest.mock('@apollo/client', () => {
  return {
    useQuery: () => ({
      loading: false,
      error: false,
      data: {
        headToHead: [
          {
            firstname: "Stan",
            lastname: "Wawrinka",
            shortname: "S.WAW",
            sex: "MAN",
            picture: {
              url: "https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
            },
            country: {
              picture: {
                url: "https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
              },
              code: "SUI"
            },
            stats: {
              rank: 21,
              points: 1784,
              weight: 81000,
              height: 183,
              age: 33,
              last: [
                true,
                true,
                true,
                false,
                true
              ]
            }
          },
          {
            firstname: "Rafael",
            lastname: "Nadal",
            shortname: "R.NAD",
            sex: "MAN",
            picture: {
              url: "https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg"
            },
            country: {
              picture: {
                url: "https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png"
              },
              code: "ESP"
            },
            stats: {
              rank: 1,
              points: 1982,
              weight: 85000,
              height: 185,
              age: 33,
              last: [
                true,
                false,
                false,
                false,
                true
              ]
            }
          }
        ]
      },
    }),
    gql: jest.fn(),
  }
})

describe('Home',  () => {
  it('should display Home', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
