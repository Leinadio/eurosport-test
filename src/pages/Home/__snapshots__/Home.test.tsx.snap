// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should display Home 1`] = `
<div>
  <h1
    className="home-title"
  >
    Players
  </h1>
  <div
    className="container"
  >
    <div
      className="cards"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          <div
            className="card-left"
          >
            <img
              alt="Stan Wawrinka"
              className="card-img"
              src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
            />
          </div>
          <div
            className="card-right"
          >
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Prenom:
              </span>
               
              Stan
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Nom:
              </span>
               
              Wawrinka
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Shortname:
              </span>
               
              S.WAW
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Age:
              </span>
               
              33
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Sexe:
              </span>
               
              MAN
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Country:
              </span>
               
              SUI
            </p>
            <img
              alt="SUI"
              className="card-country-img"
              src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
            />
          </div>
        </div>
        <div
          className="card-content"
        >
          <h1
            className="card-title"
          >
            Statistiques
          </h1>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Height:
            </span>
             
            183
            m
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Weight:
            </span>
             
            81
            Kg
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Points:
            </span>
             
            1784
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Rank:
            </span>
             
            21
          </p>
        </div>
        <div
          className="card-annexe"
        >
          <h1
            className="card-title"
          >
            Wins/Loses
          </h1>
          <div
            className="totalWinsLoses"
          >
            <p
              className="card-text"
            >
              Total Wins: 
              <span
                className="card-span totalWinsLosesValue"
              >
                0
              </span>
            </p>
            <p
              className="card-text"
            >
              Total Loses: 
              <span
                className="card-span totalWinsLosesValue"
              >
                0
              </span>
            </p>
          </div>
          <div
            className="last"
          >
            <span
              className="dot-win"
            />
            <span
              className="dot-win"
            />
            <span
              className="dot-win"
            />
            <span
              className="dot-lose"
            />
            <span
              className="dot-win"
            />
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          <div
            className="card-left"
          >
            <img
              alt="Rafael Nadal"
              className="card-img"
              src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/435121.jpg"
            />
          </div>
          <div
            className="card-right"
          >
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Prenom:
              </span>
               
              Rafael
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Nom:
              </span>
               
              Nadal
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Shortname:
              </span>
               
              R.NAD
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Age:
              </span>
               
              33
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Sexe:
              </span>
               
              MAN
            </p>
            <p
              className="card-text"
            >
              <span
                className="card-span"
              >
                Country:
              </span>
               
              ESP
            </p>
            <img
              alt="ESP"
              className="card-country-img"
              src="https://i.eurosport.com/_iss_/geo/country/flag/large/2203.png"
            />
          </div>
        </div>
        <div
          className="card-content"
        >
          <h1
            className="card-title"
          >
            Statistiques
          </h1>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Height:
            </span>
             
            185
            m
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Weight:
            </span>
             
            85
            Kg
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Points:
            </span>
             
            1982
          </p>
          <p
            className="card-text"
          >
            <span
              className="card-span"
            >
              Rank:
            </span>
             
            1
          </p>
        </div>
        <div
          className="card-annexe"
        >
          <h1
            className="card-title"
          >
            Wins/Loses
          </h1>
          <div
            className="totalWinsLoses"
          >
            <p
              className="card-text"
            >
              Total Wins: 
              <span
                className="card-span totalWinsLosesValue"
              >
                0
              </span>
            </p>
            <p
              className="card-text"
            >
              Total Loses: 
              <span
                className="card-span totalWinsLosesValue"
              >
                0
              </span>
            </p>
          </div>
          <div
            className="last"
          >
            <span
              className="dot-win"
            />
            <span
              className="dot-lose"
            />
            <span
              className="dot-lose"
            />
            <span
              className="dot-lose"
            />
            <span
              className="dot-win"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
